src/main/java/io/bf2/kafka/authorizer/CustomAclAuthorizer.java

```diff
@@ -246,6 +247,66 @@ void addAllowedListeners(Map<String, ?> configs) {
         }
     }
 
+    /**
+     * To avoid misleading users we delete principals that are explicitly targeted by CustomAclAuthorizer rules.
+     * It is possible for a user with ACL rules in Kafka's Control Plane to become a user managed by CustomAclAuthorizer,
+     * in which case we would never use or update rules for them from the Kafka Control Plane. The old rules remain,
+     * dangling, in Kafka's Control Plane and visible through kafka ACL tooling. This would mislead users because there
+     * will be a mismatch between what the rules in Kafka's Control Plane say they should be permitted to do and the truth.
+     */
+    private void deleteOrphanBindingsFromDelegate() {
+        List<AclBindingFilter> toDelete = aclPrincipals.stream().map(aclPrincipal -> {
+            AccessControlEntryFilter principalFilter = new AccessControlEntryFilter(aclPrincipal, null, AclOperation.ANY, AclPermissionType.ANY);
+            return new AclBindingFilter(ResourcePatternFilter.ANY, principalFilter);
+        }).collect(Collectors.toList());
+        if (!toDelete.isEmpty()) {
+            log.info("Orphan deletion: deleting orphaned bindings if present, using filter: {}", toDelete);
+            List<CompletableFuture<AclDeleteResult>> deletionResults = delegate.deleteAcls(null, toDelete)
+                    .stream().map(CustomAclAuthorizer::getAclDeleteResultCompletableFuture).collect(Collectors.toList());
+            CompletableFuture<Void> future = CompletableFuture.allOf(deletionResults.toArray(CompletableFuture[]::new));
+            future.whenComplete((unused, throwable) -> {
+                if (throwable == null) {
+                    log.info("Orphan deletion: deleting orphaned bindings completed successfully");
+                } else {
+                    log.error("Orphan deletion: deleting orphaned bindings completed exceptionally", throwable);
+                }
+            });
+        } else {
+            log.warn("Orphan deletion: no managed principals - skipping orphan binding deletion");
```

Comment:
warn seems excessive to me.

```suggestion
            log.info("Orphan deletion: no managed principals - skipping orphan binding deletion");
```

Reply:
No managed principals implies that the kafka instance has no owner, at least that's my understanding, so I'd expect us to log a warning about an an anomalous state.